libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/BotFrameworkHttpAdapter.cs

```diff
@@ -139,7 +139,9 @@ private async Task ConnectWebSocketAsync(IBot bot, HttpRequest httpRequest, Http
             try
             {
                 var socket = await httpRequest.HttpContext.WebSockets.AcceptWebSocketAsync().ConfigureAwait(false);
-                var requestHandler = new StreamingRequestHandler(bot, this, socket, _logger);
+                var appId = (_credentialProvider as ConfigurationCredentialProvider)?.AppId;
+                var appPassword = await _credentialProvider.GetAppPasswordAsync(appId).ConfigureAwait(false);
```

Comment:
This doesn't sound right. Doesn't work with certificate authentication, where there are not passwords.

Reply:
Changed to access the base adapter.